src/JsonSchema/SchemaFactory.php

```diff
@@ -194,8 +201,18 @@ private function buildPropertySchema(Schema $schema, string $definitionName, str
             }
 
             $subSchema = $this->buildSchema($className, $format, Schema::TYPE_OUTPUT, null, $subSchema, $serializerContext + [self::FORCE_SUBSCHEMA => true], false);
-            $propertySchema['anyOf'] = [['$ref' => $subSchema['$ref']], ['type' => 'null']];
-            // prevent "type" and "anyOf" conflict
+            if ($isCollection) {
```

Comment:
Useless `$isCollection` var as it's only used inside this foreach loop.
Also, if the first foreach loop enables the `$isCollection` variable but the same loop exists on line 200, the next `$type` will be considered as a collection even if it isn't.
```suggestion
            if ($type->isCollection()) {
```